benches/compare.rs

```diff
@@ -56,34 +65,58 @@ fn inputs(c: &mut Criterion) {
         codspeed_criterion_compat::measurement::WallTime,
     > = c.benchmark_group("Mutating Inputs");
 
-    let db = salsa::DatabaseImpl::default();
-
     group.bench_function(BenchmarkId::new("new", "InternedInput"), |b| {
-        b.iter(|| {
-            let input: InternedInput = InternedInput::new(&db, "hello, world!".to_owned());
-            interned_length(&db, input);
-        })
+        b.iter_batched_ref(
+            salsa::DatabaseImpl::default,
+            |db| {
+                let input: InternedInput = InternedInput::new(db, "hello, world!".to_owned());
+                interned_length(db, input);
+            },
+            BatchSize::SmallInput,
+        )
     });
 
     group.bench_function(BenchmarkId::new("amortized", "InternedInput"), |b| {
-        let input = InternedInput::new(&db, "hello, world!".to_owned());
-        let _ = interned_length(&db, input);
-
-        b.iter(|| interned_length(&db, input));
+        b.iter_batched_ref(
+            || {
+                let db = salsa::DatabaseImpl::default();
+                // we can't pass this along otherwise, and the lifetime is generally informational
+                let input: InternedInput<'static> =
+                    unsafe { transmute(InternedInput::new(&db, "hello, world!".to_owned())) };
```

Comment:
This is a bit annoying, the alternative I suppose is to lift out the `db` binding, assign to that, then return a reference to that but I think I prefer this over that.